evi-next-js-function-calling/src/components/ClientComponent.tsx

```diff
@@ -0,0 +1,95 @@
+"use client";
+import { VoiceProvider, ToolCall, ToolResponse, ToolError } from "@humeai/voice-react";
+import Messages from "./Controls";
+import Controls from "./Messages";
+
+const handleToolCall = async (
```

Comment:
the sdk provides a `ToolCallHandler` type that applies correct types to parameters and return values